packages/aws-cdk-lib/aws-rds/lib/proxy.ts

```diff
@@ -457,6 +457,9 @@ export class DatabaseProxy extends DatabaseProxyBase
 
     for (const secret of props.secrets) {
       secret.grantRead(role);
+      if (secret.encryptionKey !== undefined) {
```

Comment:
```suggestion
      if (secret.encryptionKey) {
```
Should be enough.

Reply:
Most certainly, this was fixed in 35089dc